src/app/reducers/local/local.ts

```diff
@@ -11,14 +13,17 @@ const initialState: State = {
 };
 
 export function localReducer(state = initialState, action: local.Action): State {
+  const len = state.closedWindows.length;
   switch (action.type) {
     case local.PUSH_CLOSED_WINDOW_TO_LOCAL:
       return {
         ...state,
-        closedWindows: [ ...state.closedWindows, action.payload.window ],
+        closedWindows: len === MAX_CLOSED_WINDOWS_LENGTH ?
+          [ ...state.closedWindows.slice(1), action.payload.window ] :
+          [ ...state.closedWindows, action.payload.window ],
+
       };
```

Comment:
@imolorhe What do you think of this change? I believe you've mentioned limiting the closed windows to 50 here https://github.com/imolorhe/altair/issues/999#issue-505677748

Reply:
Yes I did mention that. 🙂

Reply:
Question: Are you working together with @abdelrhman-arnos?

Reply:
Yes, we are.

Reply:
Nice!